### PR TITLE
Improve node label handling.

### PR DESCRIPTION
**Pull request by pjh**
- Separate LINUX_NODE_LABELS from WINDOWS_NODE_LABELS.
- Coalesce common Linux+Windows node labels.

Includes a couple other small cleanups too.

### CODE REVIEW

**Comment by pjh** [comment]
CC @yujuhong

**Comment by pjh** [comment]
Summary of the labels touched in this PR:

Linux nodes only:
  fluentd-ds-ready=true
  metadata-proxy-ready=true

Both Linux and Windows nodes:
  gke-preemptible=true
  kube-proxy-ds-ready=true

Not sure if applying gke-preemptible or kube-proxy-ds-ready to Windows nodes is ok or not, seemed probably harmless but I don't fully understand what the labels do.

**Comment by pjh** [comment]
I tested and verified that this works for both `kube-up.sh` and `e2e.go  -- --up` clusters.

File: cluster/gce/config-default.sh [hunk 1]
```diff
@@ -176,12 +169,15 @@ ENABLE_METADATA_AGENT="${KUBE_ENABLE_METADATA_AGENT:-none}"
 # Useful for scheduling heapster in large clusters with nodes of small size.
 HEAPSTER_MACHINE_TYPE="${HEAPSTER_MACHINE_TYPE:-}"
 
+LINUX_NODE_LABELS="${KUBE_NODE_LABELS:-}"
```
Comment (yujuhong) [inline cluster/gce/config-default.sh]: My original PR is to keep an env variable for common node labels, and let `LINUX/WINDOWS_NODE_LABELS` contain only OS-specific labels. There are pros and cons (e.g., code duplication, etc) of course. It's probably more important that we use these OS env variables consistently. Just a note that we should go with one approach and apply it across all env variables.

File: cluster/gce/util.sh [hunk 2]
```diff
@@ -636,37 +636,45 @@ function write-windows-node-env {
...
+# Outputs node labels common to both Linux and Windows nodes.
+function build-common-node-labels {
   local master=$1
   local node_labels=""
   if [[ "${KUBE_PROXY_DAEMONSET:-}" == "true" && "${master}" != "true" ]]; then
     # Add kube-proxy daemonset label to node to avoid situation during cluster
-    # upgrade/downgrade when there are two instances of kube-proxy running on a node.
+    # upgrade/downgrade when there are two instances of kube-proxy running on a
+    # node.
     node_labels="beta.kubernetes.io/kube-proxy-ds-ready=true"
```
Comment (yujuhong) [inline cluster/gce/util.sh]: Hmm...we actually don't want this label on windows nodes since we don't run kube-proxy as DaemonSets. I removed it in my PR.

**Comment by yujuhong** [comment]
> Not sure if applying gke-preemptible or kube-proxy-ds-ready to Windows nodes is ok or not, seemed probably harmless but I don't fully understand what the labels do.

Just saw this after I already left comments in the PR.

We should not have `kube-proxy-ds-ready` on Windows nodes...this is used in kube-proxy's node selector so that kube-proxy pods get scheduled onto the targeted nodes. We don't want that on windows.

File: cluster/gce/util.sh (hunk 2 above)
Comment (pjh) [reply cluster/gce/util.sh]: Whoops, I got the change to fluentd-ds-ready from your PR but missed kube-proxy-ds-ready. Fixed.

File: cluster/gce/config-default.sh (hunk 1 above)
Comment (pjh) [reply cluster/gce/config-default.sh]: Right, I see what you're saying. Having to split every environment variable into common+Linux+Windows is annoying... do we really need the ability for users to set all three?

I'll leave this as-is for now and revisit it in the full PR with all of the new OS env variables, to ensure consistency as you said.

File: README-GCE-Windows-kube-up.md [hunk 3]
```diff
@@ -105,11 +102,12 @@ PROJECT=${CLOUDSDK_CORE_PROJECT} go run ./hack/e2e.go  -- --up
...
 one.
 
-
 No matter what type of cluster you chose to create, the result should be a
 Kubernetes cluster with one Linux master node, two Linux worker nodes and two
-Windows worker nodes. The Linux nodes will use the `netd` CNI plugin and the
-Windows nodes will use `win-bridge`.
+Windows worker nodes. The number of nodes can be adjusted by setting the
+`NUM_LINUX_NODES` and `NUM_WINDOWS_NODES` environment variables before running
+the commands to bring up the cluster. The Linux nodes will use the `netd` CNI
```
[the remark this answers is not in the document]
Comment (pjh) [reply README-GCE-Windows-kube-up.md]: Does IP alias work for pod CIDRs when used with not-netd (with kubenet?)?

We can double-check this in the full PR and if we unnecessarily forced the Linux default to netd, we can reconsider that.

**Pull request merged**